python_chess_ai/player/user_input/terminal.py

```diff
@@ -19,15 +26,68 @@ class UserInput(UserInputInterface):
     def __init__(self):
         super().__init__()
 
-    def print_board(self, player_name, board):
-        super().print_board(player_name, board)
-        print(PLAYER_TURN_MESSAGE.format(player_name))
-        print("\n---------------")
-        print(board)
-        print("---------------\n")
+    def piece_switcher(self, piece):
+        return {
+            "K": u'\u2654',
+            "Q": u'\u2655',
+            "R": u'\u2656',
+            "B": u'\u2657',
+            "N": u'\u2658',
+            "P": u'\u2659',
+            "k": u'\u265A',
+            "q": u'\u265B',
+            "r": u'\u265C',
+            "b": u'\u265D',
+            "n": u'\u265E',
+            "p": u'\u265F'
+        }[piece]
+
+    def create_board_matrix(self, board):
+        board_fen = board.fen().split(" ")[0]
+        board_matrix = []
+
+        for row in board_fen.split("/"):
+            line = []
+            for character in row:
+                if character.isdigit():
+                    for empty in range(int(character)):
+                        line.append(" ")
+                else: 
+                    line.append(character)
+            board_matrix.append(line)
+        return board_matrix
 
     def get_move(self, legal_moves):
         super().get_move(legal_moves)
         input_msg = ASK_FOR_MOVE_MESSAGE.format(legal_moves)
         move = Tools.check_legal_input_string(legal_moves,input_msg, WRONG_INPUT_MESSAGE)
         return move
+
+    def print_alphabetical_description(self):
+        for i in range(0,9):
+            print(NUM_TO_ALPHABET[i].center(3), end='')
+        print("\n", end='')
+    
+    def print_board(self, player_name, board):
+        super().print_board(player_name, board)
+        print(PLAYER_TURN_MESSAGE.format(player_name))
+
+        board_fen = self.create_board_matrix(board)
+
+        self.print_alphabetical_description()
+        for row_index, row in enumerate(board_fen):
+            print((str(row_index + 1)).center(3), end="")
+            for field_index, chess_field in enumerate(row): 
+                field_index = bool((field_index + 1) % 2) if row_index % 2 is 0 else bool(field_index % 2)
+                colored_field = self.create_piece(chess_field, field_index)
+                print(colored_field, end='')
+            print((str(row_index + 1) + "\n").center(3), end="")
+        self.print_alphabetical_description()
+
+    def create_piece(self, character, field_index):
+        chess_piece = str(self.piece_switcher(character)) if not (character == " ") else ""
```

Comment:
Statt dem if statement hier kannst du auch oben in der Funktion returnen:
pieces_dict.get(piece," ") (oder wie auch immer du das dict nennen willst)
Dann geht das zwar nicht mehr so wie du es gemacht hast (glaube ich, vlt geht das da ja auch mit dem Komma), ist aber nach Konvention so und schöner als das hier in ner if abzufragen

Reply:
`def piece_switcher(self, piece):
        return {
            "K": u'\u2654',
            "Q": u'\u2655',
            "R": u'\u2656',
            "B": u'\u2657',
            "N": u'\u2658',
            "P": u'\u2659',
            "k": u'\u265A',
            "q": u'\u265B',
            "r": u'\u265C',
            "b": u'\u265D',
            "n": u'\u265E',
            "p": u'\u265F'
        }.get(piece, piece)`
fyi: das lässt sich auf diese weise ändern